### PR TITLE
Feat: Meta.isDeleted implement

### PR DESCRIPTION
**Pull request by jayahariv**
* add similar methods like sequence for isDeleted
* add 3 test cases for empty, single-delete, and multi-delete document.

#2232

### CODE REVIEW

File: Objective-C/Tests/QueryTest.m [hunk 1]
```diff
@@ -1759,4 +1760,105 @@ - (void) testJSONEncoding {
...
+    NSError* error;
+    NSEnumerator* rs = [q execute:&error];
+    AssertNil(error);
+    AssertEqual([[rs allObjects] count], 0u);
+}
+
+- (void) testDeleteSingleDocumentForIsDeletedExpression {
+    // save a new doc
+    NSError* error;
+    CBLMutableDocument* docmentToSave = [[CBLMutableDocument alloc] init];
```
Comment (pasin) [inline Objective-C/Tests/QueryTest.m]: docmentToSave has a typo.

File: Objective-C/Tests/QueryTest.m [hunk 2]
```diff
@@ -1759,4 +1760,105 @@ - (void) testJSONEncoding {
...
+    q = nil;
+    
+    // delete the doc
+    [self.db deleteDocument:docmentToSave error:&error];
+    AssertNil(error);
+    
+    // get no-of-deleted docs & make sure its NOT empty
+    q = [CBLQueryBuilder select: @[kDOCID]
+                                     from: [CBLQueryDataSource database: self.db]
+                                    where: [CBLQueryMeta isDeleted]];
```
Comment (pasin) [inline Objective-C/Tests/QueryTest.m]: Indentation is not off here.
